lib/TaskManager.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Created by david.winder on 04/15/2018.
+ */
+
+const util = require('util');
+const BackendClient = require('./BackendClientFactory').getBackendClient();
+const KalturaTypes = require('./kaltura-client-lib/KalturaTypes');
+const kalturaVO = require('./kaltura-client-lib/KalturaVO');
+const logger = require('../common/logger').getLogger('TaskManager');
+const config = require('../common/Configuration');
+const pushManager = require('./PushManager').PushManager;
+
+const interval = config.get('tasksPollingInterval');
+const partnerId = config.get('backendClient').partnerId;
+
+class TaskManager {
+    
+    constructor()
+    {
+        this.callbackMap = {};
+        this.mediaServerIdPromise = BackendClient.getMediaServerId();
+        this.mediaServerIdPromise.then(serverNodeId => {
+            setTimeout(()=>{this._updateList(serverNodeId);}, interval * 1000);
+        });
+    }
+    
+    register(entryServerNodeType, templateName, callback)
+    {
+        this.callbackMap[entryServerNodeType] = callback;
+        this.mediaServerIdPromise.then(serverNodeId => {
+            logger.debug("Register to Task with for " + templateName + " [" + entryServerNodeType + "] for partner: " + partnerId);
+            pushManager.registerObjectToPush(partnerId, templateName, templateName, {'serverNodeId': serverNodeId}, (...args) => this._pushMessageReceived(...args));
+            pushManager.setObjectEnabled(templateName, true);
+        });
+    }
+
+    _execIfCan(task)
+    {
+        if (this.callbackMap[task.serverType])
+            return this.callbackMap[task.serverType](task);
+        logger.info('Task with ID [' + task.id + '] cannot be execute - No callback in map');
+    }
+
+    _pushMessageReceived(msg)
+    {
+        let task = msg[0];
+        return BackendClient.getEntryServerNode(task.id).then(entryServerNode => {
+            if (entryServerNode && entryServerNode.status == KalturaTypes.KalturaEntryServerNodeStatus.TASK_PENDING)
+                return this._execIfCan(task);
+            logger.info('Task with ID [' + task.id + '] should no executed');
+        });
+    }
+
+    static _getEntryServerNodesList(serverNodeId, interval)
+    {
+        let filter = new kalturaVO.KalturaEntryServerNodeFilter();
+        filter.serverNodeIdEqual = serverNodeId;
+        filter.statusIn = KalturaTypes.KalturaEntryServerNodeStatus.TASK_PENDING;
+        filter.createdAtLessThanOrEqual = - interval;
```

Comment:
does this works?

didn't you said that you will ask everything lower than now-X  so there won't be race?